src/test/suite/extension.test.ts

```diff
@@ -15,4 +16,11 @@ suite("Extension Test Suite", () => {
       "Hello, {world! [This] is a (test.)}"
     );
   });
+
+  test("Special characters in object keys", () => {
+    assert.strictEqual(
+      formatDiagnosticMessage("Type 'string' is not assignable to type '{ 'abc*bc': string; }'.").match(/<code>(.*?)<\/code>/)?.[0],
+      "<code>{ 'abc*bc': string; }</code>"
+    );
+  });
```

Comment:
We'll have to format these cases so they'll no longer be an unstyled code block so the test has to be changed too